app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java

```diff
@@ -33,25 +34,43 @@ public void init() {
         ButterKnife.bind(this, getActivity());
     }
 
+
     @Override
     public void setListeners() {
 
         vLogInButton.setOnClickListener(new View.OnClickListener() {
             @OnClick
             public void onClick(View v) {
-                if (vLogInEmail.getText().toString().isEmpty() && vLogInPassword.getText().toString().isEmpty())
-                    Toast.makeText(requireContext(), "Los campos de correo electronico y contraseña se encuentran vacios", Toast.LENGTH_SHORT).show();
-                else if (vLogInEmail.getText().toString().isEmpty())
-                    Toast.makeText(requireContext(), "Ingrese correo electronico", Toast.LENGTH_SHORT).show();
-                else if (!android.util.Patterns.EMAIL_ADDRESS.matcher(vLogInEmail.getText().toString()).matches())
-                    Toast.makeText(requireContext(), "Ingrese correo electronico valido", Toast.LENGTH_SHORT).show();
-                else if (vLogInPassword.getText().toString().isEmpty())
-                    Toast.makeText(requireContext(), "Ingrese Contraseña valida", Toast.LENGTH_SHORT).show();
-                else if (vLogInPassword.getText().toString().length() < 6)
-                    Toast.makeText(requireContext(), "La contraseña debe tener mas de 6 caracteres", Toast.LENGTH_SHORT).show();
-                else
-                    Toast.makeText(requireContext(), "Login presionado", Toast.LENGTH_SHORT).show();
+                if (areValidFields())
+                    getPresenter().doLogin(vLogInEmail.getText().toString(), vLogInPassword.getText().toString());
             }
+
         });
     }
+
+    @Override
+    public void onLoginSuccess() {
+        getPresenter().setPreferences(getContext(), vLogInEmail.getText().toString());
+        Intent intent = new Intent(getActivity(), HomeActivity.class);
+        intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+        startActivity(intent);
+    }
+
+
+    private boolean areValidFields() {
```

Comment:
this should be done in the presenter

Reply:
I rename now the method areValidFields() to ValidationFields in LoginPresenter